### PR TITLE
Script for Running CUB Benchmarks

### PR DESCRIPTION
**Pull request by senior-zero**
## Description

closes https://github.com/NVIDIA/cccl_private/issues/151

This PR adds a script for running all CUB benchmarks and reporting results in the format similar to the one used in `thrust/internal/scripts/eris_perf.py`. More details are provided in `cub/docs/benchmarking.rst`. 

## Checklist

- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cub/docs/benchmarking.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+*************************************
+
+This file contains instrutions on how to run all CUB benchmarks using CUB tuning infrastructure.
+
+.. code-block:: bash
+
+    pip3 install --user fpzip pandas scipy
+    git clone https://github.com/NVIDIA/cccl.git
+    mkdir build; cd build
+    cmake .. -DCCCL_ENABLE_THRUST=OFF\
```
Comment (miscco) [inline cub/docs/benchmarking.rst:11]: suggestion: We should rather tell cmake to create the build dir as we do in the ci scripts
```suggestion
    cmake -S .. -B build
             -DCCCL_ENABLE_THRUST=OFF\
```

File: cub/docs/benchmarking.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+             -DCUB_ENABLE_DIALECT_CPP11=OFF\
+             -DCUB_ENABLE_DIALECT_CPP14=OFF\
+             -DCUB_ENABLE_DIALECT_CPP17=ON\
+             -DCUB_ENABLE_DIALECT_CPP20=OFF\
+             -DCUB_ENABLE_RDC_TESTS=OFF\
+             -DCUB_ENABLE_BENCHMARKS=YES\
+             -DCUB_ENABLE_TUNING=YES\
+             -DCMAKE_BUILD_TYPE=Release\
+             -DCMAKE_CUDA_ARCHITECTURES="89;90"
+    ../cub/benchmarks/scripts/run.py
```
Comment (miscco) [inline cub/docs/benchmarking.rst:23]:
```suggestion
    cub/benchmarks/scripts/run.py
```

**Review by miscco — APPROVED** [review]

**Review by elstehle — APPROVED** [review]

File: cub/docs/benchmarking.rst (hunk 1 above)
Comment (jrhemstad) [reply cub/docs/benchmarking.rst:11]: Note (non-blocking): let's add presets for CUB benchmarks as part of https://github.com/NVIDIA/cccl/pull/324. 

In an ideal future world, we won't need anyone to do explicit cmake invocations with anything other than a particular preset (and maybe GPU archs).

File: cub/docs/benchmarking.rst (hunk 2 above)
Comment (senior-zero) [reply cub/docs/benchmarking.rst:23]: It's executed inside `build` directory:
```
mkdir build; cd build
```

**Pull request merged**